scripts/quests/outlands/You_Call_That_a_Knife.lua

```diff
@@ -0,0 +1,210 @@
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:event(318)
+                    end
+                end
+            },
+
+            ['Pula_Rhatti'] = quest:event(269, 0, xi.item.SANDFISH):replaceDefault(),
+
+            ['_6y9'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(128)
+                    end
+                end
+            },
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    if option == 511 then
+                        quest:setVar(player, 'Prog', 1)
+                        player:confirmTrade()
+                    end
+                end,
+
+                [128] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and
+                not player:hasKeyItem(xi.ki.NONBERRYS_KNIFE)
+        end,
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Mhebi_Juhbily'] = quest:event(129):replaceDefault(),
+
+            ['Pula_Rhatti'] = quest:event(269, 0, xi.item.SANDFISH):replaceDefault(),
+
+            ['Vah_Keshura'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(130)
+                    else
+                        return quest:event(131):replaceDefault()
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [130] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end
+            },
+        },
+
+        [xi.zone.TEMPLE_OF_UGGALEPIH] =
+        {
+            ['Chef_Nonberry'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { xi.item.TONBERRY_BOARD }) then
+                        return quest:progressEvent(27)
+                    else
+                        return quest:progressEvent(28)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [27] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, xi.ki.NONBERRYS_KNIFE)
+                end,
+
+                [28] = function(player, csid, option, npc)
+                    player:confirmTrade()
```

Comment:
This is not consuming the item as it should

Reply:
Fixed, running all trade slots through confirmSlot before confirmTrade is called. Thanks for spotting this!